test/functionalTest/cases/4286_update_format_for_notification_messages/update_format_for_notification_messages.test

```diff
@@ -0,0 +1,196 @@
+echo "====================================================================================="
+payload='
+{
+  "subject": {
+    "entities": [
+      {
+        "id": "E1"
+      }
+    ],
+    "condition": {
+      "attrs": [ "A1" ]
+    }
+  },
+  "notification": {
+    "httpCustom": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [ "A1" ],
+    "attrsFormat": "simplifiedNormalized"
+  },
+  "expires": "2050-04-05T14:00:00.00Z"
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "02. Create NGSIv2 subscription for E1/A1 with attrsFormat set to simplifiedKeyValues"
+echo "===================================================================================="
+payload='
+{
+  "subject": {
+    "entities": [
+      {
+        "id": "E1"
+      }
+    ],
+    "condition": {
+      "attrs": [ "A1" ]
+    }
+  },
+  "notification": {
+    "httpCustom": {
+      "url": "http://localhost:'$LISTENER_PORT'/notify"
+    },
+    "attrs": [ "A1" ],
+    "attrsFormat": "simplifiedKeyValues"
+  },
+  "expires": "2050-04-05T14:00:00.00Z"
+}'
+orionCurl --url /v2/subscriptions --payload "$payload"
+echo
+echo
+
+
+echo "03. Create entity E1/T1"
+echo "======================="
+payload='{
+  "id": "E1",
+  "type": "T1",
+  "A1": {
+    "value": 1,
+    "type": "Number"
+  }
+}'
+orionCurl --url /v2/entities --payload "${payload}"
+echo
+echo
+
+
+echo "04. Dump accumulator to see notifications"
+echo "========================================="
+accumulatorDump
+echo
+echo
+
+
+--REGEXPECT--
+01. Create NGSIv2 subscription for E1/A1 with attrsFormat set to simplifiedNormalized
+=====================================================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+02. Create NGSIv2 subscription for E1/A1 with attrsFormat set to simplifiedKeyValues
+====================================================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+03. Create entity E1/T1
+=======================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/entities/E1?type=T1
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+04. Dump accumulator to see notifications
+=========================================
+POST http://localhost:REGEX(\d+)/notify
+Fiware-Servicepath: /
+Content-Length: 72
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Ngsiv2-Attrsformat: simplifiedNormalized
+Host: localhost:REGEX(\d+)
+Accept: application/json
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36}; cbnotif=[12345])
+
+[
+    {
+        "A1": {
+            "metadata": {},
+            "type": "Number",
+            "value": 1
+        },
+        "id": "E1",
+        "type": "T1"
+    }
+]
```

Comment:
Entity shouldn't be contained in an array, for the sake of simplicity and to mimics the update format:

```suggestion
{
    "A1": {
        "metadata": {},
        "type": "Number",
        "value": 1
    },
    "id": "E1",
    "type": "T1"
}
```

Reply:
> Entity shouldn't be contained in an array, for the sake of simplicity and to mimics the update format:

As per my understanding this output comes from https://github.com/telefonicaid/fiware-orion/pull/4298/files#diff-3ff97d5dc39ed89271e80e2c3cbd327799f05b7d0846e6747ba1b9dfbd801436R93  which is in the form of array. For changing the response we need to change contextElementResponseVector and lots of functions need to be modify. 
Could you please suggest any other way of doing it?

Reply:
Since the removal of initial notification functionality time ago, notifications contains always one entity. Thus, you can call the render of the first element of the contextElementResponseVector vector. Something like this:

```
    out += contextElementResponseVector[0].toJson(renderFormat, attrsFilter, blacklist, metadataFilter, replacementsP);
```

(Not fully sure of the exact syntax as I don't have my development environment at hand right now, but I guess you get the idea :)

Reply:
> Since the removal of initial notification functionality time ago, notifications contains always one entity. Thus, you can call the render of the first element of the contextElementResponseVector vector. Something like this:
> 
> ```
>     out += contextElementResponseVector[0].toJson(renderFormat, attrsFilter, blacklist, metadataFilter, replacementsP);
> ```
> 
> (Not fully sure of the exact syntax as I don't have my development environment at hand right now, but I guess you get the idea :)

Thank you for suggestion I have updated my PR in https://github.com/telefonicaid/fiware-orion/pull/4298/commits/9c2c1156a57bf3d4a80a418c1240609e80b2ac1f